src/WebJobs.Script/Utility.cs

```diff
@@ -533,8 +533,29 @@ private static bool ContainsFunctionWithWorkerRuntime(IEnumerable<FunctionMetada
             return false;
         }
 
-        public static bool CheckAppOffline(string scriptPath)
+        public static async Task MarkContainerDisabled(ILogger logger)
         {
+            logger.LogInformation("Setting container instance offline");
+            var disableContainerFilePath = Path.Combine(Path.GetTempPath(), ScriptConstants.DisableContainerFileName);
+            if (!FileUtility.FileExists(disableContainerFilePath))
+            {
+                await FileUtility.WriteAsync(disableContainerFilePath, "This container instance is offline");
+            }
+        }
+
+        public static bool IsContainerDisabled()
+        {
+            return FileUtility.FileExists(Path.Combine(Path.GetTempPath(), ScriptConstants.DisableContainerFileName));
+        }
+
+        public static bool CheckAppOffline(IEnvironment environment, string scriptPath)
```

Comment:
AppOffline and Container disabled are two separate states. Keep them separate

Reply:
We will have logs to know why the app was put offline and also i think they are just different ways of putting the app offline. so prefer to keep it this way to avoid more code churn.